### PR TITLE
Django view

### PR DESCRIPTION
**Pull request by shlior7**
Create the view for the app

based on the Django initialization made by #8

### CODE REVIEW

**Comment by yuvalk** [comment]
commit msgs should be more meaningful, explaining why you did things
also consider maybe squashing them a bit if it make sense

File: board/forms.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if 'country' in self.data:
+            try:
+                country_id = int(self.data.get('country'))
+                self.fields['city'].queryset = City.objects.filter(
+                    country_id=country_id).order_by('name')
+            except (ValueError, TypeError):
+                pass  # invalid input from the client; ignore and fallback to empty City queryset
+        elif self.instance.pk:
+            self.fields['city'].queryset = self.instance.country.city_set.order_by(
+                'name')
```
Comment (shlior7) [inline board/forms.py:24]: Setting the city dropdown in accordance to the country chosen in `self.data` of the form

File: board/views.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class PersonUpdateView(UpdateView):
+    model = Person
+    form_class = PersonForm
+    success_url = reverse_lazy('person_changelist')
+
+
+def load_cities(request):
+    country_id = request.GET.get('country')
+    cities = City.objects.filter(country_id=country_id).order_by('name')
+    return render(request, 'board/city_dropdown_list_options.html', {'cities': cities})
```
Comment (shlior7) [inline board/views.py:29]: loading the cities by first a GET request of the country from the form and then filtering by that `country_id` and rendering the `city_dropdown_list_options.html` with the appropriate cities

File: board/views.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+from django.views.generic import ListView, CreateView, UpdateView
+from django.urls import reverse_lazy
+
+from .models import Person, City
+from .forms import PersonForm
+
+
+class PersonListView(ListView):
+    model = Person
+    context_object_name = 'people'
```
Comment (shlior7) [inline board/views.py:11]: for referencing the persons list from the db

File: board/templates/board/person_list.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          <td><a href="{% url 'person_change' person.pk %}">{{ person.name }}</a></td>
+          <td>{{ person.birthdate }}</td>
+          <td>{{ person.country.name }}</td>
+          <td>{{ person.city.name }}</td>
+          <td>{{ person.job }}</td>
+          <td>{{ person.income }}</td>
+          <td>{{ person.expenses }}</td>
+          <td>{{ person.kids }}</td>
+        </tr>
+      {% empty %}
```
Comment (shlior7) [inline board/templates/board/person_list.html:37]: showing the persons list from db as `people` in a table

File: board/templates/board/person_list.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        <tr>
+          <td colspan="4">No person in the database. <a href="{% url 'person_add' %}">Add the first person</a>.</td>
```
Comment (shlior7) [inline board/templates/board/person_list.html:39]: adding a person reference to the url path `person_add` directed to `/add`

File: board/templates/board/person_form.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+      $.ajax({
+        url: url,
+        data: {
+          'country': countryId
+        },
+        success: function (data) {
+          $("#id_city").html(data);
+        }
+      });
```
Comment (shlior7) [inline board/templates/board/person_form.html:30]: importing Jquery for using ajax and requesting the cities through the `ajax_load_cities` url when the country id in the form change

**Comment by shlior7** [comment]
> commit msgs should be more meaningful, explaining why you did things
also consider maybe squashing them a bit if it make sense
Added comments explaining important implementation parts, also I don't see where squashing would be necessary as the commits are set in a way to better explain each part of the pr, zooming in the critical parts of this implementation (Model, Form, View, Url, Html etc... )

File: board/migrations/0002_add_initial_data.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# Generated by Django 2.0.1 on 2018-01-28 19:30
+
+from django.db import migrations
+
+
+def add_initial_data(apps, schema_editor):
+    Country = apps.get_model('board', 'Country')
+    City = apps.get_model('board', 'City')
+
+    israel = Country.objects.create(name='Isreal')
```
Comment (adiabramovitch) [inline board/migrations/0002_add_initial_data.py:10]: you have a typo here: Israel

**Comment by adiabramovitch** [comment]
You have a few commits that are not relevant, you should rebase because those commits were already merged. 
moreover, you have commits that are repetitive, you should squash the ones that deal with the same files or logic. 
I recommend looking at this manual for interactive rebasing: 
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by Yoavsc129** [comment]
We need to update some of the files in order to match the requirements of demo 2. specificaly the person class and everything else that derives from it.

**Pull request closed without merging**